.github/workflows/tests.yml

```diff
@@ -1,10 +1,20 @@
 name: Tests
 
-on: [push]
+on:
+  push:
+    branches:
+      - "**" # all branches
+      - "!master" # except master
+      - "!develop" # except develop
```

Comment:
I think it's important that we run checks against these branches because they enable tests to catch issues with logical merge conflicts. Particularly, `on: push` for a PR is generally going to run checks against the commits in the PR, not the merge commit for that PR back into its target branch; including `develop` ensures that we get test coverage of the merged result.

We could consider using `on: pull_request` to cover PRs instead (which runs against merge commits rather than direct commits), but it has the disadvantage that you then don't start running tests until after you've finished writing up a PR description, and it's nice in practice to be able to start seeing test results earlier.